gui/wxpython/dbmgr/base.py

```diff
@@ -3865,10 +3865,10 @@ def OnChangeLayer(self, event):
         """Layer number of layer to be deleted is changed"""
         try:
             layer = int(event.GetString())
-        except:
+        except Exception:
             try:
                 layer = self.mapDBInfo.layers.keys()[0]
-            except:
+            except Exception:
```

Comment:
>     1. AttributeError: This can occur if self.mapDBInfo does not have the attribute layers, or if layers does not have the method keys().

We should generally expect that the objects have the attributes we expect, so AttributeError is something which should cause traceback in general.

>     2. TypeError: This can occur if self.mapDBInfo.layers.keys() does not return a sequence type (like list or tuple). In Python 3, dict.keys() returns a view object, not a list, so you can't index it directly.

Given that we are running Python 3, if layers is a dict (I did not check), this always fails then and needs to be fixed. This seems to be something to exclude from this PR and fix in a separate PR. This PR is too large to handle any changes which are not strictly exception specifications. 

>     3. IndexError: This can occur if the sequence returned by self.mapDBInfo.layers.keys() is empty. Trying to access the first element of an empty sequence will raise this exception.

I think this looks like the exception which is supposed to be caught here. Given the previous, it seems that changing it here will cause an error, so solving this together with the `keys()` call in that separate PR is needed.